libbeat/outputs/shipper/shipper.go

```diff
@@ -156,28 +178,33 @@ func (c *shipper) Publish(ctx context.Context, batch publisher.Batch) error {
 		})
 	}
 
+	st.Dropped(dropped)
 	c.log.Debugf("%d events converted to protobuf, %d dropped", len(grpcEvents), dropped)
 
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
 	defer cancel()
-	_, err := c.client.PublishEvents(ctx, &sc.PublishRequest{
+	resp, err := c.client.PublishEvents(ctx, &sc.PublishRequest{
 		Events: grpcEvents,
 	})
 
-	if err != nil {
-		if status.Code(err) == codes.ResourceExhausted {
-			c.log.Warn("shipper's queue is full, more events cannot be accepted")
-			batch.Cancelled()
-		} else {
-			batch.Retry()
-		}
-		return fmt.Errorf("failed to publish the batch to the shipper: %w", err)
+	if status.Code(err) != codes.OK {
+		// if the response is nil, it means none of the events made it through due to a severe failure
```

Comment:
I'm a bit confused by this comment... You're checking the error and not using the response at all. Then why mention it?